### PR TITLE
feat(DataTable): add support for table row statuses

### PR DESCRIPTION
**Pull request by booc0mtaco**
- component to support standard EDS statuses per row
- update tests and snapshots (new story for comparison)

### Test Plan:

- [x] Wrote/updated [automated tests](https://czi.atlassian.net/wiki/x/Hbl1H)
- [ ] CI tests / new tests are not applicable
- [ ] Manually tested my changes, and here are the details:

### CODE REVIEW

File: src/components/DataTable/DataTable.stories.tsx [hunk 1]
```diff
@@ -700,6 +705,93 @@ export const Grouping: StoryObj<Args> = {
...
+    tableStyle: 'border',
+    rowStyle: 'lined',
+    size: 'sm',
+  },
+  render: (args) => {
+    const columns = [
+      columnHelper.accessor(DataTable.__StatusColumnId__, {
+        header: () => <DataTable.StatusHeaderCell />,
+        cell: (info) => <DataTable.StatusCell status={info.getValue()} />,
+        size: 32,
```
Comment (booc0mtaco) [inline src/components/DataTable/DataTable.stories.tsx:729]: TODO add / use some token here

File: src/components/DataTable/DataTable.tsx [hunk 2]
```diff
@@ -159,129 +176,141 @@ export function DataTable<T>({
...
-              <DataTableRow key={headerGroup.id}>
-                {headerGroup.headers.map((header) => (
-                  <th
-                    className={styles['data-table__header-cell-container']}
-                    colSpan={header.colSpan}
-                    key={header.id}
-                    style={{
-                      width: `${header.getSize()}px`,
-                    }}
+    <DataTableContext.Provider value={{ size }}>
```
Comment (booc0mtaco) [inline src/components/DataTable/DataTable.tsx:179]: Most of this diff is whitespace changes. Use the ⚙️ above to update the view to hide whitespace changes.

File: src/components/DataTable/DataTable.tsx [hunk 3]
```diff
@@ -159,129 +176,141 @@ export function DataTable<T>({
...
+            )}
+            <DataTableHeader>
+              {table?.getHeaderGroups().map((headerGroup) => (
+                <DataTableRow key={headerGroup.id}>
+                  {headerGroup.headers.map((header) => {
+                    // TODO-AH: figure out better way to detect a non-default column size
+                    const columnWidth =
+                      header.getSize() !== 150
+                        ? `${header.getSize()}px`
+                        : undefined;
```
Comment (booc0mtaco) [inline src/components/DataTable/DataTable.tsx:237]: Need to investigate how TanStack Table sets the column width. `.getSize` is always defined, and defaults to 150px. Marking this value as special here b/c all sizings will be based on token unit sizes (16 * x) and 150 doesn't land on one of those boundaries (... 144, 160, 176, ...)

File: src/index.ts [hunk 4]
```diff
@@ -61,6 +61,11 @@ export type { ButtonProps } from './components/Button';
 export type { LinkProps } from './components/Link';
 export type { PageNotificationProps } from './components/PageNotification';
 export type { AppNotificationProps } from './components/AppNotification';
+export type {
+  DataTableProps,
+  DataTableRowProps,
+  DataTableWithStatus,
```
Comment (booc0mtaco) [inline src/index.ts:67]: Utility type for augmenting table model data with the details of the status type in the type object.

**Review by timzchang — APPROVED** [review]
looks good to me!

**Pull request merged**